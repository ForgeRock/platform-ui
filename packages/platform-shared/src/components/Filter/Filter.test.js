/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { mount, flushPromises } from '@vue/test-utils';
import FilterModal from './index';

describe('Filter.test.js', () => {
  let wrapper;

  const filtersStringArray = [
    'filter1',
    'filter2',
    'filter3',
    'filter4',
    'filter5',
  ];

  const filtersObjectArray = [
    {
      text: 'Filter 1',
      value: 'objFilter1',
    },
    {
      text: 'Filter 2',
      value: 'objFilter2',
    },
    {
      text: 'Filter 3',
      value: 'objFilter3',
    },
    {
      text: 'Filter 4',
      value: 'objFilter4',
    },
    {
      text: 'Filter 5',
      value: 'objFilter5',
    },
  ];

  const filtersCategoriesArray = [
    {
      categories: ['Cat1', 'Cat3'],
      text: 'Filter 1',
      value: 'objFilter1',
    },
    {
      categories: ['Cat2', 'Cat4', 'Cat5'],
      text: 'Filter 2',
      value: 'objFilter2',
    },
    {
      categories: ['Cat1'],
      text: 'Filter 3',
      value: 'objFilter3',
    },
    {
      categories: ['Cat1', 'Cat2', 'Cat3', 'Cat4', 'Cat5'],
      text: 'Filter 4',
      value: 'objFilter4',
    },
    {
      categories: ['Cat2'],
      text: 'Filter 5',
      value: 'objFilter5',
    },
  ];

  const categories = ['Cat1', 'Cat2', 'Cat3', 'Cat4', 'Cat5'];

  const factory = (props) => {
    const component = mount(FilterModal, {
      attachTo: document.body,
      global: {
        mocks: {
          $t: () => {},
          $tc: () => {},
        },
      },
      props,
    });
    component.vm.$root.$emit('bv::show::modal', 'FilterModal');
    return component;
  };

  it('Loads the filter with string array filters', async () => {
    wrapper = factory({ filters: filtersStringArray });
    wrapper.vm.show = true;

    // No categories, so don't render button nav
    const buttonContainer = wrapper.find('.filter-type-radio');
    expect(buttonContainer.exists()).toBeFalsy();

    const renderedOptions = [];
    wrapper.findAll('#filter-list .multiselect__element').filter((filterItem) => renderedOptions.push(filterItem.text()));
    expect(renderedOptions).toEqual(expect.arrayContaining(filtersStringArray));
  });

  it('Loads the filter with object array filters', async () => {
    wrapper = factory({ filters: filtersObjectArray });
    wrapper.vm.show = true;

    // No categories, so don't render button nav
    const buttonContainer = wrapper.find('.filter-type-radio');
    expect(buttonContainer.exists()).toBeFalsy();

    const renderedOptions = [];
    wrapper.findAll('#filter-list .multiselect__element').filter((filterItem) => renderedOptions.push(filterItem.text()));
    const filterTextLabels = filtersObjectArray.map((filter) => filter.text);
    expect(renderedOptions).toEqual(expect.arrayContaining(filterTextLabels));
  });

  it('Loads the filter with categories array filters', async () => {
    wrapper = factory({ filters: filtersCategoriesArray, categories });
    wrapper.vm.show = true;
    wrapper.vm.filterTypeSelected = 'categories';
    await flushPromises();

    // We have categories
    const buttonContainer = wrapper.find('.filter-type-radio');
    expect(buttonContainer.exists()).toBeTruthy();

    const renderedOptions = [];
    wrapper.findAll('#categories-list .multiselect__element').filter((filterItem) => renderedOptions.push(filterItem.text()));
    expect(renderedOptions).toEqual(expect.arrayContaining(categories));
  });

  it('Loads the filter with multiple category array filters', async () => {
    const activeCategories = ['Cat1', 'Cat5'];
    const activeFilters = ['objFilter1', 'objFilter2', 'objFilter3', 'objFilter4'];
    wrapper = factory({
      filters: filtersCategoriesArray, categories, activeCategories, activeFilters,
    });
    wrapper.vm.show = true;
    wrapper.vm.filterTypeSelected = 'categories';
    await flushPromises();

    // Check active categories
    const selectedRenderedCategories = [];
    wrapper.findAll('#categories-list .multiselect__tag').filter((filterItem) => selectedRenderedCategories.push(filterItem.text().trim()));
    expect(selectedRenderedCategories).toEqual(activeCategories);

    wrapper.vm.filterTypeSelected = 'filters';
    await flushPromises();

    // Check active filters
    const selectedRenderedFilters = [];
    wrapper.findAll('#filter-list .multiselect__tag').filter((filterItem) => selectedRenderedFilters.push(filterItem.text().trim()));
    expect(selectedRenderedCategories).toEqual(activeCategories);
  });

  it('Loads with active string array filters', async () => {
    const activeFilters = ['filter3', 'filter5'];

    wrapper = factory({ filters: filtersStringArray, activeFilters });
    wrapper.vm.show = true;
    await flushPromises();

    const selectedRenderedOptions = [];
    wrapper.findAll('#filter-list .multiselect__tag').filter((filterItem) => selectedRenderedOptions.push(filterItem.text().trim()));
    expect(selectedRenderedOptions).toEqual(activeFilters);
  });

  it('Loads with active object array filters', async () => {
    const activeFilters = [
      {
        text: 'Filter 3',
        value: 'objFilter3',
      },
      {
        text: 'Filter 5',
        value: 'objFilter5',
      },
    ];

    wrapper = factory({ filters: filtersObjectArray, activeFilters });
    wrapper.vm.show = true;
  });

  it('Emits pending values when options are selected', async () => {
    wrapper = factory({ filters: filtersStringArray });
    wrapper.vm.show = true;
    await flushPromises();

    // Check emitted value
    wrapper.findAll('#filter-list .multiselect__element .multiselect__option')[1].trigger('click');
    wrapper.findAll('#filter-list .multiselect__element .multiselect__option')[3].trigger('click');
    await flushPromises();
    expect(wrapper.emitted()).toHaveProperty('update');
    expect(wrapper.emitted().update).toHaveLength(2);
    expect(wrapper.emitted().update[1][0]).toEqual(wrapper.vm.pendingFilters);
    expect(wrapper.emitted().update[1][1]).toEqual(wrapper.vm.pendingCategories);
  });

  it('Emits pending values when category options are selected', async () => {
    wrapper = factory({ filters: filtersCategoriesArray, categories });
    wrapper.vm.show = true;
    wrapper.vm.filterTypeSelected = 'categories';
    await flushPromises();

    // Check emitted value
    wrapper.findAll('#categories-list .multiselect__element .multiselect__option')[0].trigger('click');
    wrapper.findAll('#categories-list .multiselect__element .multiselect__option')[2].trigger('click');
    await flushPromises();
    expect(wrapper.emitted()).toHaveProperty('update');
    expect(wrapper.emitted().update).toHaveLength(2);
    expect(wrapper.emitted().update[1][0]).toEqual(wrapper.vm.pendingFilters);
    expect(wrapper.emitted().update[1][1]).toEqual(wrapper.vm.pendingCategories);
  });

  it('Clears the filters', () => {
    const activeFilters = ['filter3', 'filter5'];
    wrapper = factory({ filters: filtersStringArray, activeFilters });
    wrapper.vm.show = true;

    // Clear the filters
    wrapper.vm.clearFilters();

    const selectedRenderedOptions = [];
    wrapper.findAll('#filter-list .multiselect__tag').filter((filterItem) => selectedRenderedOptions.push(filterItem.text().trim()));

    expect(selectedRenderedOptions).toEqual([]);
    expect(wrapper.vm.pendingFilters).toEqual([]);
  });

  it('Clears the categories', async () => {
    const activeCategories = ['Cat1', 'Cat5'];
    const activeFilters = ['objFilter1', 'objFilter2', 'objFilter3', 'objFilter4'];
    wrapper = factory({
      filters: filtersCategoriesArray, categories, activeCategories, activeFilters,
    });
    wrapper.vm.show = true;

    // Clear the categories & filters
    wrapper.vm.clearCategories();

    // Categories
    const selectedRenderedCategories = [];
    wrapper.findAll('#filter-list .multiselect__tag').filter((filterItem) => selectedRenderedCategories.push(filterItem.text().trim()));

    wrapper.vm.filterTypeSelected = 'filters';
    await wrapper.vm.$nextTick;

    // Filters
    const selectedRenderedFilters = [];
    wrapper.findAll('#filter-list .multiselect__tag').filter((filterItem) => selectedRenderedFilters.push(filterItem.text().trim()));

    expect(selectedRenderedFilters).toEqual([]);
    expect(selectedRenderedCategories).toEqual([]);
    expect(wrapper.vm.pendingCategories).toEqual([]);
    expect(wrapper.vm.pendingFilters).toEqual([]);
  });
});
