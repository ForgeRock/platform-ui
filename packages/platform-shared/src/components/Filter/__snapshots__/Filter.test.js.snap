// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter.test.js Loads the filter with categories array filters: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) (2)",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_SCOPED_SLOTS) vm.$scopedSlots has been removed. Use vm.$slots instead.
  Details: https://v3-migration.vuejs.org/breaking-changes/slots-unification.html",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) (2)",
    "
",
    " at <BFormRadioGroup",
    "modelValue="filters"",
    "onUpdate:modelValue=fn",
    "aria-describedby=undefined",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) (3)",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (3)",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_SCOPED_SLOTS) (2)",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) (4)",
    "
",
    " at <BFormRadio",
    "key="BV_option_0"",
    "aria-describedby=undefined",
    "disabled=false",
    " ...",
    ">",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_SCOPED_SLOTS) (3)",
    "
",
    " at <BFormRadio",
    "key="BV_option_0"",
    "aria-describedby=undefined",
    "disabled=false",
    " ...",
    ">",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "aria-invalid" with v-bind value \`false\` will render aria-invalid="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BFormRadioGroup",
    "aria-describedby=undefined",
    "options=",
    [
      {
        "text": undefined,
        "value": "filters",
      },
      {
        "text": undefined,
        "value": "categories",
      },
    ],
    "button-variant="outline-secondary"",
    " ...",
    ">",
    "
",
    " at <BFormGroup",
    "key=0",
    "class="filter-type-radio"",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      {
        "categories": [
          "Cat1",
          "Cat3",
        ],
        "text": "Filter 1",
        "value": "objFilter1",
      },
      {
        "categories": [
          "Cat2",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 2",
        "value": "objFilter2",
      },
      {
        "categories": [
          "Cat1",
        ],
        "text": "Filter 3",
        "value": "objFilter3",
      },
      {
        "categories": [
          "Cat1",
          "Cat2",
          "Cat3",
          "Cat4",
          "Cat5",
        ],
        "text": "Filter 4",
        "value": "objFilter4",
      },
      {
        "categories": [
          "Cat2",
        ],
        "text": "Filter 5",
        "value": "objFilter5",
      },
    ],
    "categories=",
    [
      "Cat1",
      "Cat2",
      "Cat3",
      "Cat4",
      "Cat5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;

exports[`Filter.test.js Loads the filter with string array filters: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (2)",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_V_MODEL) v-model usage on component has changed in Vue 3. Component that expects to work with v-model should now use the "modelValue" prop and emit the "update:modelValue" event. You can update the usage and then opt-in to Vue 3 behavior on a per-component basis with \`compatConfig: { COMPONENT_V_MODEL: false }\`.
  Details: https://v3-migration.vuejs.org/breaking-changes/v-model.html",
    "
",
    " at <FrField",
    "modelValue=",
    [],
    "onUpdate:modelValue=fn",
    "label=undefined",
    " ...",
    ">",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation COMPONENT_FUNCTIONAL) Functional component <BButton> should be defined as a plain function in Vue 3. The "functional" option has been removed. NOTE: Before migrating to use plain functions for functional components, first make sure that all async components usage have been migrated and its compat behavior has been disabled.
  Details: https://v3-migration.vuejs.org/breaking-changes/functional-components.html",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (3)",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) vm.$listeners has been removed. In Vue 3, parent v-on listeners are included in vm.$attrs and it is no longer necessary to separately use v-on="$listeners" if you are already using v-bind="$attrs". (Note: the Vue 3 behavior only applies if this compat config is disabled)
  Details: https://v3-migration.vuejs.org/breaking-changes/listeners-removed.html",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (4)",
    "
",
    " at <MultiSelect",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "name="filter-list"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation WATCH_ARRAY) (5)",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "type="multiselect"",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "name="filter-list"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "appear" with v-bind value \`false\` will render appear="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "type="multiselect"",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "name="filter-list"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "persisted" with v-bind value \`false\` will render persisted="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <Transition",
    "name="multiselect__loading"",
    ">",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "type="multiselect"",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "name="filter-list"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_ENUMERATED_COERCION) Enumerated attribute "spellcheck" with v-bind value \`false\` will render the value as-is instead of coercing the value to "true" in Vue 3. Always use explicit "true" or "false" values for enumerated attributes. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_ENUMERATED_COERCION: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <VueMultiselect",
    "key=0",
    "id="floatingLabelInput5"",
    "type="multiselect"",
    " ...",
    ">",
    "
",
    " at <InputLayout",
    "description=""",
    "id="floatingLabelInput5"",
    "errors=",
    [],
    " ...",
    ">",
    "
",
    " at <MultiSelect",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "name="filter-list"",
    " ...",
    ">",
    "
",
    " at <FrField",
    "label=undefined",
    "options=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "class="mb-4"",
    " ...",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation INSTANCE_LISTENERS) (2)",
    "
",
    " at <BButton",
    "class="p-0"",
    "variant="link"",
    "onClick=fn<bound clearFilters>",
    ">",
    "
",
    " at <FilterComponent",
    "filters=",
    [
      "filter1",
      "filter2",
      "filter3",
      "filter4",
      "filter5",
    ],
    "ref="VTU_COMPONENT"",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
